### DISCUSSION TITLE
Comparison to other coroutines libraries

### DISCUSSION THREAD

**Discussion by rbeeli** [resetius/coroio#36] [General]
Hi,

Interesting, lightweight library. How does it compare to other C++20 coroutines based networking libraries? E.g.
* https://github.com/lewissbaker/cppcoro
* https://github.com/chriskohlhoff/asio/
* https://github.com/alibaba/PhotonLibOS
* https://github.com/alibaba/async_simple

And probably many more. What are the pros- and cons of this implementation?
What are the long-term goals?

**Comment by resetius** [comment]
Thank you for the question. I'll take a look at these implementations. The main idea behind my library was to create a small, lightweight I/O library whose code I fully control and can use in my projects. This allows me to make adjustments as necessary when issues arise. Initially, I developed this library specifically for implementing my own raft server (see the adjacent repository).

As for long-term goals, I do not plan to significantly expand the library's functionality. I will be adding utilities on top of the library's core and fixing bugs. However, it's important to note that the library is not used in any enterprise-like projects, so development may be relatively slow.
